### ISSUE TITLE
Unable to insert into table with materialized view

### ISSUE THREAD

**Issue by lustefaniak** [scylladb/scylla#3009]
*Installation details*
Scylla version (or git commit hash): 2.0.1
Cluster size: 1
OS (RHEL/CentOS/Ubuntu/AWS AMI): Docker with `--experimental 1`

I try to execute those queries:
```
CREATE KEYSPACE IF NOT EXISTS respondent_overlaps  WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'};
CREATE TABLE IF NOT EXISTS respondent_overlaps.respondent_overlaps (
respondent_id varchar,
hour int,
timeseries_key varchar,
value text,
encoding_version int,
PRIMARY KEY (respondent_id, hour, timeseries_key)
);
CREATE MATERIALIZED VIEW respondent_overlaps.respondent_id_by_hour AS
SELECT respondent_id, hour, timeseries_key FROM respondent_overlaps.respondent_overlaps
PRIMARY KEY (hour,respondent_id, timeseries_key);
INSERT INTO respondent_overlaps.respondent_overlaps (respondent_id, hour, timeseries_key, value, encoding_version) VALUES('ent_id', 123, 'key', 'value', 1);
```

On the last one I get error:
```
WriteTimeout: Error from server: code=1100 [Coordinator node timed out waiting for replica nodes' responses] message="Operation timed out for respondent_overlaps.respondent_overlaps - received only 0 responses from 1 CL=ONE." info={'received_responses': 0, 'required_responses': 1, 'consistency': 'ONE'}
```

In Scylla docker logs I see:
```
ERROR 2017-11-22 23:16:14,454 [shard 1] storage_proxy - exception during mutation write to 172.17.0.2: exceptions::syntax_exception (line 1:75 no viable alternative at input 'ALLOW')
```

Is there something I'm doing wrong and I shouldn't be able to do when defining materialized view?
Is there any way to see what query was generated and failed parsing?

**Comment by lustefaniak** [comment]
Ok, looks like the reason is missing WHERE statement in materialized view creation. But I don't think this is expected behaviour when only feedback is error in server log.

**Comment by tzach** [comment]
The highlight of the Scylla MV feature is here:
http://www.scylladb.com/2017/07/27/materialized-views-preview-scylla-2-0/

We will soon add a MV section to Scylla Doc, in the mean time, the CQL is compatible with Apache Cassandra 
http://cassandra.apache.org/doc/latest/cql/mvs.html

**Comment by gleb-cloudius** [comment]
I agree that this error is not the right response to a wrong syntax. @duarten ?

**Comment by duarten** [comment]
Right, lets leave this open. I'm probably missing validation of the where clause when creating the materialized view.

**Comment by gleb-cloudius** [comment]
On Thu, Nov 23, 2017 at 10:31:41AM +0000, Duarte Nunes wrote:
> Right, lets leave this open. I'm probably missing validation of the where clause when creating the materialized view.
> 
Do you know how this missing validation can cause the timeout error?

--
			Gleb.

**Comment by lustefaniak** [comment]
The error which I pasted happens when `INSERT` is executing, that's also a moment when error in logs appears.

**Comment by duarten** [comment]
@gleb-cloudius Nop. Maybe `exceptions::syntax_exception` is not being propagated back to the coordinator?

**Comment by gleb-cloudius** [comment]
On Thu, Nov 23, 2017 at 11:35:12AM +0000, Duarte Nunes wrote:
> @gleb-cloudius Nop. Maybe `exceptions::syntax_exception` is not being propagated back to the coordinator? 
> 
Ah, this may be the case indeed. Writes do not propagate errors since
they use rpc message without reply.

--
			Gleb.

**Comment by duarten** [comment]
This was fixed by da110d612e4d690d80292256d94238ee7d22db17.
